armi/bookkeeping/db/layout.py

```diff
@@ -851,18 +851,11 @@ def replaceNonsenseWithNones(data: numpy.ndarray, paramName: str) -> numpy.ndarr
             if isNone[i].all():
                 result[i] = None
             elif isNone[i].any():
-                # TODO: This is not symmetric with the replaceNonesWithNonsense impl.
-                # That one assumes that Nones apply only at the highest dimension, and
-                # that the lower dimensions will be filled with the magic None value.
-                # Non-none entries below the top level fail to coerce to a serializable
-                # numpy array and would raise an exception when trying to write. TL;DR:
-                # this is a dead branch until the replaceNonesWithNonsense impl is more
-                # sophisticated.
```

Comment:
I guess `replaceNonesWithNonsense` is now sufficiently sophisticated? 🤷‍♂️

Reply:
I mean, imagine you need to "replace Nonsense with Nones", WHY would you want to use a function called "replace Nones with Nonsense"?  

That would be EXTRA confusing to explain to people.

Also... if that one function did both things I would want to break it up into two smaller functions that only do one thing each.